Cargo.toml

```diff
@@ -19,6 +19,7 @@ ring = "0.17"
 rustls = "0.23"
 rustls-webpki = "0.102"
 serial_test = "3"
+tempfile = "3.5"
```

Comment:
Would you mind folding the `Cargo.lock` update into this commit? I think they should be done in sync so that this commit builds cleanly with `--locked`